src/main/webapp/js/instructorFeedbackEdit/questionRubric.es6

```diff
@@ -0,0 +1,256 @@
+                        name="rubricDesc-${newRowNumber - 1}-${cols}">
+                </textarea>
+            </td>`;
+        rubricRowBodyFragments += rubricRowFragment;
+    }
+
+    // Create new rubric row
+    const newRubricRow =
+        `<tr id="rubricRow-${questionNum}-${newRowNumber - 1}">
+            <td>
+                <div class="col-sm-12 input-group">
+                    <span class="input-group-addon btn btn-default rubricRemoveSubQuestionLink-${questionNum}"
+                            id="rubricRemoveSubQuestionLink-${questionNum}-${newRowNumber - 1}"
+                            onclick="removeRubricRow(${newRowNumber - 1}, ${questionNum})"
+                            onmouseover="highlightRubricRow(${newRowNumber - 1}, ${questionNum}, true)"
+                            onmouseout="highlightRubricRow(${newRowNumber - 1}, ${questionNum}, false)">
+                        <span class="glyphicon glyphicon-remove"></span>
+                    </span>
+                    <textarea class="form-control" rows="3" id="rubricSubQn-${questionNum}-${newRowNumber - 1}"
+                            name="rubricSubQn-${newRowNumber - 1}">
+                    </textarea>
+                </div>
+            </td>
+            ${rubricRowBodyFragments}
+        </tr>`;
+
+    // Row to insert new row after
+    const lastRow = $(`#rubricEditTable-${questionNum} tr:last`);
+    $(newRubricRow).insertAfter(lastRow);
+
+    // Increment
+    $(`#rubricNumRows-${questionNum}`).val(newRowNumber);
+
+    if ($(questionId).attr('editStatus') === 'hasResponses') {
+        $(questionId).attr('editStatus', 'mustDeleteResponses');
+    }
+}
+
+function addRubricCol(questionNum) {
+    const questionId = `#form_editquestion-${questionNum}`;
+
+    const numberOfRows = parseInt($(`#rubricNumRows-${questionNum}`).val(), 10);
+    const numberOfCols = parseInt($(`#rubricNumCols-${questionNum}`).val(), 10);
+
+    const newColNumber = numberOfCols + 1;
+
+    // Insert header <th>
+    const rubricHeaderFragment =
+        `<th class="rubricCol-${questionNum}-${newColNumber - 1}">
+            <div class="input-group">
+                <input type="text" class="col-sm-12 form-control" value=""
+                        id="rubricChoice-${questionNum}-${newColNumber - 1}"
+                        name="rubricChoice-${newColNumber - 1}">
+                <span class="input-group-addon btn btn-default rubricRemoveChoiceLink-${questionNum}"
+                        id="rubricRemoveChoiceLink-${questionNum}-${newColNumber - 1}"
+                        onclick="removeRubricCol(${newColNumber - 1}, ${questionNum})"
+                        onmouseover="highlightRubricCol(${newColNumber - 1}, ${questionNum}, true)"
+                        onmouseout="highlightRubricCol(${newColNumber - 1}, ${questionNum}, false)">
+                    <span class="glyphicon glyphicon-remove"></span>
+                </span>
+            </div>
+        </th>`;
```

Comment:
> (Random wild idea: Shall we migrate the HTML to Pug/Jade? Then we can compile those back to HTML just before insertion into a page. Snippets of Pug are way more readable than snippets of HTML, and there are a lot of these sorts of snippets scattered all over the codebase.)

I think doing this will just further fragment (pun intended) the jstl, jsp, html and js we already have.